packages/lwc-language-server/src/base-indexer.ts

```diff
@@ -0,0 +1,21 @@
+import * as path from 'path';
+import * as fsExtra from 'fs-extra';
+export default class BaseIndexer {
+    readonly workspaceRoot: string;
+    constructor(attributes: { workspaceRoot: string }) {
+        this.workspaceRoot = path.resolve(attributes.workspaceRoot);
+    }
+
+    sfdxConfig(root: string) {
+        const filename: string = path.join(root, 'sfdx-project.json');
+        const data: string = fsExtra.readFileSync(filename).toString();
+
+        return JSON.parse(data);
```

Comment:
`sfdx-project.json` is a file customers might change often and we've seen cases where they have invalid json. We should add some error handling for this scenario and throw an error message that users can act on.